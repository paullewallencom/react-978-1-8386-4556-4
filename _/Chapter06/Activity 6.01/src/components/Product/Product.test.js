import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { Product } from "./index";

afterEach(cleanup);

const product = {
  id: 1,
  name: "Expresso",
  price: 5.5,
  summary:
    "Galão, grinder mocha, filter plunger pot siphon mug variety est caramelization. Galão cortado medium latte organic aroma dripper lungo extraction crema. Bar variety spoon a fair trade filter iced. To go mocha wings irish kopi-luwak acerbic doppio kopi-luwak viennese saucer aroma arabica.",
  tags: ["black"]
};

it("renders product", () => {
  const { asFragment } = render(<Product product={product} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders the right information", () => {
  const { container } = render(<Product product={product} />);
  expect(container.querySelector("h2").innerHTML).toBe(product.name);
  expect(container.querySelector("strong").innerHTML).toBe("$" + product.price);
});

it("product information is visible when the button is clicked", () => {
  const { container } = render(<Product product={product} />);
  expect(container.firstChild.lastChild.firstChild).toBeNull();
  fireEvent.click(container.querySelector("button"));
  expect(container.firstChild.lastChild.firstChild.innerHTML).toBe(
    product.summary
  );
});
