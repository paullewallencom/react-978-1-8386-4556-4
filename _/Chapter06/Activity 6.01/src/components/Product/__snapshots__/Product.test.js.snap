// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product 1`] = `
<DocumentFragment>
  <div
    class="product"
  >
    <div
      class="product_header"
    >
      <h2>
        Expresso
      </h2>
      <div>
        <button>
          +
        </button>
      </div>
    </div>
    <strong>
      $5.5
    </strong>
    <div />
  </div>
</DocumentFragment>
`;
